### PR TITLE
Python 3 Explicit support

### PR DESCRIPTION
**Pull request by platinummonkey**
- Explicit Python3 support, still support python2
- runs `black` formatting tool across code, fixes a few nit styling issues
- encapsulates tests in unittest.TestCase

### CODE REVIEW

**Review by andrewkandalaft — APPROVED** [review]
Sweet, py3 only version we need to make sure we don't accidentally update to this 😅

**Comment by platinummonkey** [comment]
yeah technically, the last PR is py3 compatible, I'm just cleaning up

https://github.com/DataDog/credis/releases/tag/1.0.6%2Bdd.1 for the last release compatible

**Comment by platinummonkey** [comment]
@andrewkandalaft keeping py2 support for now :) updated

File: setup.py [hunk 1]
```diff
@@ -1,29 +1,38 @@
 #!/usr/bin/env python
-try:
-    from setuptools import setup
```
Comment (andrewkandalaft) [inline setup.py:3]: I think you need to keep this try/except for py2 compatibility

Comment (platinummonkey) [reply setup.py:3]: it worked for me locally :D 

```bash
(venv2) ➜  venv2/bin/python --version
Python 2.7.17
(venv2) ➜  venv2/bin/python setup.py install
/.../Versions/2.7/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'install_requires'
  warnings.warn(msg)
running install
running build
running build_py
running build_ext
skipping 'credis/base.c' Cython extension (up-to-date)
skipping 'credis/geventpool.c' Cython extension (up-to-date)
running install_lib
running install_egg_info
Removing /..../venv2/lib/python2.7/site-packages/credis-2.0.0-py2.7.egg-info
Writing /..../venv2/lib/python2.7/site-packages/credis-2.0.0-py2.7.egg-info
(venv2) ➜  echo $?
0
```

**Review by andrewkandalaft — APPROVED** [review]

**Pull request merged**